src/build.h

```diff
@@ -0,0 +1,381 @@
+
```

Comment:
functions in header needs to be weak symbol, I mean,`inline` must be added, or move these functions to .cc and make a header file for that.